From c3a006f789fe9303102ffcac38cb84b6b2ad9e69 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Thu, 6 Feb 2020 15:35:56 +0200
Subject: [PATCH 09/14] Replace no error code '0' for SMS with '-1'

  * from
  https://github.com/LineageOS/android_packages_apps_Messaging/commit/c35f2b019f5c6a1f0f04654e23a93fecd8520da4

	modified:   src/com/android/messaging/receiver/SendStatusReceiver.java
	modified:   src/com/android/messaging/receiver/SmsReceiver.java
---
 src/com/android/messaging/receiver/SendStatusReceiver.java | 2 +-
 src/com/android/messaging/receiver/SmsReceiver.java        | 3 ++-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/com/android/messaging/receiver/SendStatusReceiver.java b/src/com/android/messaging/receiver/SendStatusReceiver.java
index fc0e8c9..64b0560 100644
--- a/src/com/android/messaging/receiver/SendStatusReceiver.java
+++ b/src/com/android/messaging/receiver/SendStatusReceiver.java
@@ -51,7 +51,7 @@ public class SendStatusReceiver extends BroadcastReceiver {
     public static final String EXTRA_PART_ID = "partId";
     public static final String EXTRA_SUB_ID = "subId";
 
-    public static final int NO_ERROR_CODE = 0;
+    public static final int NO_ERROR_CODE = -1;
     public static final int NO_PART_ID = -1;
 
     @Override
diff --git a/src/com/android/messaging/receiver/SmsReceiver.java b/src/com/android/messaging/receiver/SmsReceiver.java
index 859beff..4b3c5cc 100644
--- a/src/com/android/messaging/receiver/SmsReceiver.java
+++ b/src/com/android/messaging/receiver/SmsReceiver.java
@@ -186,7 +186,8 @@ public final class SmsReceiver extends BroadcastReceiver {
             return;
         }
 
-        final int errorCode = intent.getIntExtra(EXTRA_ERROR_CODE, 0);
+        final int errorCode =
+                intent.getIntExtra(EXTRA_ERROR_CODE, SendStatusReceiver.NO_ERROR_CODE);
         // Always convert negative subIds into -1
         int subId = PhoneUtils.getDefault().getEffectiveIncomingSubIdFromSystem(
                 intent, EXTRA_SUB_ID);
-- 
2.11.0

